### PR TITLE
Fix Store.State.propType to work with optional/isRequired

### PR DESCRIPTION
**Pull request by elierotenberg**

### CODE REVIEW

File: lib/Store.js [hunk 1]
```diff
@@ -57,19 +57,7 @@ class State {
...
-   * ...
-   * static propTypes = {
-   *   foo: Store.State.propType(React.PropTypes.array),
-   * };
-   * ...
-   */
-  static propType(type) {
+  static validatePropType(type) {
     /**
      * Validates a {@link State} prop.
```
Comment (eTorAken) [inline lib/Store.js]: I think this doc should be above `validatePropType` declaration, no?

**Comment by Hadrien-DELAITRE** [comment]
This works! as expected :)
Just add jsDoc here: https://github.com/elierotenberg/react-nexus/blob/fix-propTypes-isRequired/lib/Store.js#L60, and it's ok!

**Comment by Hadrien-DELAITRE** [comment]
Need last review before merging. @elierotenberg @antca @eTorAken

**Comment by elierotenberg** [comment]
:+1:

**Comment by antca** [comment]
:+1:

**Pull request merged**